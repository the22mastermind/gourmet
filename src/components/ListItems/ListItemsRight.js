import React from 'react';
import { Caption } from 'react-native-paper';
import { StyleSheet, Text, View } from 'react-native';

const styles = StyleSheet.create({
  container: {
    justifyContent: 'space-evenly',
    alignItems: 'flex-end',
    marginLeft: 12,
  },
  bottom: {
    fontSize: 14,
    fontWeight: '700',
    color: '#a4c49a',
    textTransform: 'capitalize',
  },
});

const ListItemsRight = ({ top, bottom, ...rest }) => (
  <View style={styles.container} testID="list-item-right-wrapper">
    <Caption testID="list-item-right-top" {...rest}>{top}</Caption>
    <Text testID="list-item-right-bottom" {...rest} style={styles.bottom}>
      {bottom}
    </Text>
  </View>
);

export default ListItemsRight;
