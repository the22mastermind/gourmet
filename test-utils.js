import React from 'react';
import { render } from '@testing-library/react-native';
import { Provider as PaperProvider } from 'react-native-paper';
import { AuthProvider } from './src/context/AuthProvider';
import { DataProvider } from './src/context/DataProvider';
import { AlertProvider } from './src/context/AlertProvider';
import theme from './src/context/theme';

const AllTheProviders = ({ children }) => (
  <AuthProvider>
    <DataProvider>
      <AlertProvider>
        <PaperProvider theme={theme}>
          {children}
        </PaperProvider>
      </AlertProvider>
    </DataProvider>
  </AuthProvider>
);

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react-native';

export { customRender as render };
