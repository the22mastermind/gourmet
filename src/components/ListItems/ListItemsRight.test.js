import React from 'react';
import { render } from 'test-utils';
import ListItemsRight from './ListItemsRight';

const props = {
  top: 'large',
  bottom: '$4.5',
  color: '#333333',
};

describe('<ListItemsRight /> component', () => {
  it('renders <ListItemsRight /> component correctly', () => {
    const screen = render(<ListItemsRight {...props} />);
  
    expect(screen.getByTestId('list-item-right-wrapper')).toBeDefined();
    expect(screen.getByTestId('list-item-right-top').props.children).toBe('large');
    expect(screen.getByTestId('list-item-right-bottom').props.children).toBe('$4.5');
  });
  
  it('matches <ListItemsRight /> component snapshot', async () => {
    const screen = render(<ListItemsRight {...props} />);
  
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
