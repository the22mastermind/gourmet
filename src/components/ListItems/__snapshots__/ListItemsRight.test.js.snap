// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItemsRight /> component matches <ListItemsRight /> component snapshot 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "justifyContent": "space-evenly",
        "marginLeft": 12,
      }
    }
    testID="list-item-right-wrapper"
  >
    <Text
      color="#333333"
      style={
        Array [
          Object {
            "color": "#333333",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
          },
          Array [
            Object {
              "color": "rgba(51, 51, 51, 0.54)",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "fontSize": 12,
                "letterSpacing": 0.4,
                "lineHeight": 20,
                "marginVertical": 2,
              },
              undefined,
            ],
          ],
        ]
      }
      testID="list-item-right-top"
    >
      large
    </Text>
    <Text
      color="#333333"
      style={
        Object {
          "color": "#a4c49a",
          "fontSize": 16,
          "fontWeight": "700",
          "textTransform": "capitalize",
        }
      }
      testID="list-item-right-bottom"
    >
      $4.5
    </Text>
  </View>
</View>
`;
